### PR TITLE
WIP: Supporting functions for array symbolics

### PR DESCRIPTION
**Pull request by shashi**
builds on #215

### CODE REVIEW

**Comment by shashi** [comment]
https://github.com/JuliaParallel/DistributedArrays.jl/issues/4 is a good list of operations to wrap.

File: src/types.jl [hunk 1]
```diff
@@ -125,9 +125,11 @@ means the variable is a function with the type signature X -> Y where
 """
 struct Sym{T} <: Symbolic{T}
     name::Symbol
+    metadata::Dict{Type, Any}
```
[the remark this answers is not in the document]
Comment (shashi) [reply src/types.jl]: Now this field is `nothing` by default, and `ImmutableDict{DataType, Any}` if there's metadata.

File: src/arrays.jl [hunk 2]
```diff
@@ -0,0 +1,146 @@
+const SymArray = Symbolic{<:AbstractArray}
```
[the remark this answers is not in the document]
Comment (YingboMa) [reply src/arrays.jl:1]: It's easier to call `eltype` and `ndims` than matching the type params

**Comment by shashi** [comment]
cc @YingboMa check out all the smilarterm changes lol, we should make that the default maybe?? But you were the one who switched it back to using `promote_symtype`.

**Pull request merged**